src/txn/txn.c

```diff
@@ -572,11 +572,14 @@ __wt_txn_update_oldest(WT_SESSION_IMPL *session, uint32_t flags)
 
         /* Output a verbose message about long-running transactions,
          * but only when some progress is being made. */
+        current_id = __wt_atomic_loadv64(&txn_global->current);
+        WT_ASSERT(session, WT_TXNID_LE(oldest_id, current_id));
```

Comment:
This assertion is guaranteed to always be true as we're inside the txn_global write lock. I think we can keep it though.